### PR TITLE
Allow simulating epJSON files and running regressions

### PR DESCRIPTION
**Pull request by Myoldmopar**

### CODE REVIEW

File: epregressions/builds/base.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def get_idfs_in_dir(idf_dir: Path) -> Set[Path]:
         idf_path = Path(idf_dir)
         all_idfs_absolute_path = list(idf_path.rglob('*.idf'))
         all_idfs_absolute_path.extend(list(idf_path.rglob('*.imf')))
+        all_idfs_absolute_path.extend(list(idf_path.rglob('*.epJSON')))
```
Comment (Myoldmopar) [inline epregressions/builds/base.py:46]: This change allows the epJSON files to show up in the GUI for selection

File: epregressions/energyplus.py [hunk 2]
```diff
@@ -94,57 +94,58 @@ def execute_energyplus(e_args: ExecutionArguments):
...
-                with open('SLABSurfaceTemps.TXT') as f:
-                    append_text = f.read()
-                with open('in.idf', 'a') as f:
-                    f.write("\n%s\n" % append_text)
-                os.remove('SLABINP.TXT')
-                os.remove('GHTIn.idf')
-                os.remove('SLABSurfaceTemps.TXT')
-                os.remove('SLABSplit Surface Temps.TXT')
-                os.remove('SlabGHT.idd')
+        # Run ExpandObjects and process as necessary, but not for epJSON files!
```
Comment (Myoldmopar) [inline epregressions/energyplus.py:97]: Only run ExpandObjects for IDF-based files.  Running it on an epJSON file results in an existing, but invalid expanded.idf file, which confuses later code.

File: epregressions/energyplus.py [hunk 3]
```diff
@@ -171,8 +172,11 @@ def execute_energyplus(e_args: ExecutionArguments):
 
         # Execute EnergyPlus
         try:
+            command_line = energyplus
+            if os.path.exists('in.epJSON'):
+                command_line = energyplus + ' in.epJSON'
             std_out += subprocess.check_output(
-                energyplus, shell=True, stdin=subprocess.DEVNULL, stderr=subprocess.PIPE
+                command_line, shell=True, stdin=subprocess.DEVNULL, stderr=subprocess.PIPE
```
Comment (Myoldmopar) [inline epregressions/energyplus.py:179]: By default, with no arguments, EnergyPlus will try to run `in.idf`.  It's a different story that we should be using the command line interface here, but for now, we need to at least tell it to use the `in.epJSON` file.

File: epregressions/runtests.py [hunk 4]
```diff
@@ -218,8 +218,10 @@ def run_build(self, build_tree):
                         shutil.copy(
                             full_file_name, os.path.join(build_tree['build_dir'], this_test_dir, this_entry.basename)
                         )
+            elif full_input_file_path.endswith('.epJSON'):
```
Comment (Myoldmopar) [inline epregressions/runtests.py:221]: Add option for epJSON file extension.

File: epregressions/tests/resources/dummy.energyplus.py [hunk 5]
```diff
@@ -22,7 +23,10 @@
 import json
 import sys
 
-with open('in.idf') as f_idf:
+file_name = 'in.idf'
+if os.path.exists('in.epJSON'):
+    file_name = 'in.epJSON'
+with open(file_name) as f_idf:
```
Comment (Myoldmopar) [inline epregressions/tests/resources/dummy.energyplus.py:29]: Had to modify the dummy EnergyPlus script to allow it to recognize a possible `in.epJSON` file.

File: epregressions/tests/test_runtests.py [hunk 6]
```diff
@@ -1944,6 +1944,73 @@ def test_both_success_no_diffs_dd_only(self):
...
+        self.assertEqual(1, len(diff_results.entries_by_file))
+        results_for_file = diff_results.entries_by_file[0]
+        # it should be named according to what we listed above
+        self.assertEqual('my_file', results_for_file.basename)
+        # it should have succeeded in both base and mod cases
+        self.assertEqual(EndErrSummary.STATUS_SUCCESS, results_for_file.summary_result.simulation_status_case1)
+        self.assertEqual(EndErrSummary.STATUS_SUCCESS, results_for_file.summary_result.simulation_status_case2)
+        # it should have created a test directory and dropped the summaries there
+        results_dir = diff_results.results_dir_a
+        self.assertIn('DDOnly', results_dir)
```
Comment (Myoldmopar) [inline epregressions/tests/test_runtests.py:2012]: New unit test for finding and running epJSON files.

File: epregressions/tk_window.py [hunk 7]
```diff
@@ -93,7 +93,7 @@ def __init__(self):
 
         # high level GUI configuration
         self.root.geometry('1000x600')
-        self.root.resizable(width=1, height=1)
+        self.root.resizable(width=True, height=True)
```
Comment (Myoldmopar) [inline epregressions/tk_window.py:96]: IDE warnings revealed some bad arguments and weird imports, so this file has some miscellaneous cleanups.

**Comment by Myoldmopar** [comment]
@mjwitte @mitchute @dareumnam @jmythms @jmarrec if any of you use the regression tool on your platform of choice and would be able to test this branch out, I would appreciate it.  I already kicked off a branch of EnergyPlus using this branch to test that CI runs it fine.  I'm actually more worried about local usage.  If you are able to run a regression on the RefBldg...epJSON file successfully, that would be enough evidence.

No pressure to do it at all, and if you do, then no rush, enjoy the weekend and let me know next week.

**Comment by mitchute** [comment]
While you're in here... 😄 

Could you make the drop-down boxes for run configuration and reporting frequency work in dark mode on Mac, like we did in #84?

Screen Shot 2022-04-01 at 3 49 01 PM

And I should say, that the options are visible, just not the selected option.
Screen Shot 2022-04-01 at 3 50 36 PM

**Comment by mitchute** [comment]
(and yes, I'll test it)

**Comment by mitchute** [comment]
RefBldg epJSON works great locally on Mac. 👍

**Comment by Myoldmopar** [comment]
Thanks @mitchute !  I can't get my debugging environment working right now on Mac, so I'm not going to be trying to do the button style changes until I get that fixed.  I'm going to go ahead and merge this and make a new release so that other folks can pip install -U to get the update from PyPi.  In case something is found after that is tested then I can follow-up, but I think it will be fine.

**Comment by dareumnam** [comment]
Thanks for the new release. I'll get the update and test.

**Comment by Myoldmopar** [comment]
Thanks dareum, something actually went wrong with the packaging. I'll let you know when it's ready and how to verify you have the right version.

**Pull request merged**